src/error.rs

```diff
@@ -19,7 +19,7 @@ pub enum Error {
         received: u16,
     },
     /// Something timed out.
-    Timeout,
+    Timeout(crate::timer_factory::TimeoutKind),
```

Comment:
Nit: please move the import to the top of the file and write this as `Timeout(TimeoutKind)`